src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -173,23 +170,25 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
                 return;
             }
 
-            package.IsChanging = true;
-
-            if (_selectedTab == SelectedTab.InstalledPackages || _selectedTab == SelectedTab.LoadedPackages) {
-                IsLoading = true;
-            }
+            BeforeLoadUnload(package);
 
             await UpdateImplAsync(package);
+            await AfterLoadUnloadAsync(package);
+        }
 
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                ReplaceItems(_installedPackages);
-                IsLoading = false;
-            } else if (_selectedTab == SelectedTab.LoadedPackages) {
-                ReplaceItems(_loadedPackages);
-                IsLoading = false;
+        private async Task ReplaceItemsAsync() {
+            switch (_selectedTab) {
+                case SelectedTab.AvailablePackages:
+                    await ReplaceItemsAsync(_availablePackages);
+                    break;
+                case SelectedTab.InstalledPackages:
+                    await ReplaceItemsAsync(_installedPackages);
+                    break;
+                case SelectedTab.LoadedPackages:
+                    await ReplaceItemsAsync(_loadedPackages);
```

Comment:
>ReplaceItemsAsync [](start = 26, length = 17)

I don't think `ReplaceItemsAsync` should change `IsLoading` to false - this method has no information about the context and can be called by ReloadItemsAsync from callback of the main session while packages on the opened tab are still loaded from the background session